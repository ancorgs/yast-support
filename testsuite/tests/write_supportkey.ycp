{

    include "testsuite.ycp";

    include "support/supportio.ycp";

    map WRITE_ok =
    $[ "support":
		$[ "key": true ]
	];

    map WRITE_notok =
    $[ "support":
		$[ "key": false ]
	];

    map keydata =
    $[ "name1": "Washington",
       "name2": "Irwing",
       "salutation": "male",
       "company": "M&M",
       "street": "",
       "zip": "",
       "country": "Italy",
       "state": "",
       "city": "",
       "email": "WI@IW.jh",
       "key": "WI-CATCH22-Yo-Yo"
    ];

    // keydata map succesfully written to the file
	TEST (``(write_supportkey(keydata)), [ $[], WRITE_ok, $[] ], nil);
    DUMP ("----------------------");

    // unsuccesfull write to the file
	TEST (``(write_supportkey(keydata)), [ $[], WRITE_notok, $[] ], nil);
    DUMP ("----------------------");

    keydata =
    $[ "name1": "Washington",
       "name2": "Irwing",
       "salutation": "male",
       "email": "WI@IW.jh",
       "key": "WI-CATCH22-Yo-Yo"
       ];

    // keydata map is not full - for mising keys the empty string is written
	TEST (``(write_supportkey(keydata)), [ $[], WRITE_ok, $[] ], nil);
}
