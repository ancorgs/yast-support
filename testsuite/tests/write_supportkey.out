Write	.support.key.KEY_NAME1 "Washington" true
Write	.support.key.KEY_NAME2 "Irwing" true
Write	.support.key.KEY_SALUTATION "male" true
Write	.support.key.KEY_COMPANY "M&M" true
Write	.support.key.KEY_STREET "" true
Write	.support.key.KEY_ZIP "" true
Write	.support.key.KEY_COUNTRY "Italy" true
Write	.support.key.KEY_STATE "" true
Write	.support.key.KEY_CITY "" true
Write	.support.key.KEY_EMAIL "WI@IW.jh" true
Write	.support.key.KEY_CODE "WI-CATCH22-Yo-Yo" true
Write	.support.key nil true
Return	true
Dump	----------------------
Write	.support.key.KEY_NAME1 "Washington" false
Write	.support.key.KEY_NAME2 "Irwing" false
Write	.support.key.KEY_SALUTATION "male" false
Write	.support.key.KEY_COMPANY "M&M" false
Write	.support.key.KEY_STREET "" false
Write	.support.key.KEY_ZIP "" false
Write	.support.key.KEY_COUNTRY "Italy" false
Write	.support.key.KEY_STATE "" false
Write	.support.key.KEY_CITY "" false
Write	.support.key.KEY_EMAIL "WI@IW.jh" false
Write	.support.key.KEY_CODE "WI-CATCH22-Yo-Yo" false
Write	.support.key nil false
Return	false
Dump	----------------------
Write	.support.key.KEY_NAME1 "Washington" true
Write	.support.key.KEY_NAME2 "Irwing" true
Write	.support.key.KEY_SALUTATION "male" true
Write	.support.key.KEY_COMPANY "" true
Write	.support.key.KEY_STREET "" true
Write	.support.key.KEY_ZIP "" true
Write	.support.key.KEY_COUNTRY "" true
Write	.support.key.KEY_STATE "" true
Write	.support.key.KEY_CITY "" true
Write	.support.key.KEY_EMAIL "WI@IW.jh" true
Write	.support.key.KEY_CODE "WI-CATCH22-Yo-Yo" true
Write	.support.key nil true
Return	true
